### PR TITLE
Qwen Image / Image Edit support

### PR DESCRIPTION
**Pull request by choppu**
* qwen image / qwen image edit support
* download script
* multiimage editing support
* added optional fusion for Qwen Image (Edit) LoRAs
* updated README.md

### CODE REVIEW

File: thenoise/dit/qwen_image/utils.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+    The 7B model is constructed on ``meta`` (via ``init_empty_weights``) so only
+    the checkpoint weights are materialized; the loader assigns them directly and
+    moves the model to ``device``.
+    """
+    config = Qwen2_5_VLConfig(**json.loads(QWEN2_5_VL_CONFIG_JSON))
+    with init_empty_weights():
+        model = Qwen2_5_VLForConditionalGeneration._from_config(config)
+        # The conditional-generation head is not used for embedding extraction.
+        model.lm_head = torch.nn.Identity()
```
Comment (bitgamma) [inline thenoise/dit/qwen_image/utils.py]: this doesn't follow the pattern other models use. The usual pattern is `del model.lm_head; replace_linears(model)`

File: tests/conftest.py [hunk 2]
```diff
@@ -302,11 +303,35 @@ def wrapped_fp8_tensor():
             "model.diffusion_model.final_layer.adaLN_modulation.1.weight",
         ],
     ),
+    "qwen_image": (
+        QwenImageModel,
+        [
+            "img_in.weight",
+            "txt_in.weight",
+            "time_text_embed.1.weight",
+            "layers.0.img_attn.qkv.weight",
```
Comment (bitgamma) [inline tests/conftest.py]: the `layers.*` values do not exist in real checkpoint. Verify against a real checkpoint and only include realistic names in the fixture

File: thenoise/dit/qwen_image/models.py [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+        # Key-padding mask: image tokens are always valid, text tokens follow the mask.
+        if encoder_hidden_states_mask is not None:
+            attention_mask = torch.cat(
+                [
+                    torch.ones((encoder_hidden_states_mask.shape[0], seq_img), device=encoder_hidden_states_mask.device, dtype=torch.bool),
+                    encoder_hidden_states_mask.to(torch.bool),
+                ],
+                dim=1,
+            )
+            attention_mask = attention_mask[:, None, None, :]
```
Comment (bitgamma) [inline thenoise/dit/qwen_image/models.py]: The mask is always constructed and always passed as attn_mask=. An explicit bool mask excludes FA/efficient SDPA backends and forces the math path. At B=1 with no padding the mask is all-ones; if not mask.all(): … else attn_mask=None (and skipping the torch.ones image half) is worth trying to check if there is any performance improvement

File: thenoise/dit/qwen_image/models.py [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+    num_layers: int = 60,
+) -> QwenImageTransformer2DModel:
+    """Load the Qwen-Image DiT via the central quant-aware loader.
+
+    ``init_empty_weights`` (not ``torch.device("meta")``) keeps ``pos_freqs``/
+    ``neg_freqs`` as real plain tensors, since ``model.to(device)`` won't move them.
+    """
+    with init_empty_weights():
+        model = create_model(zero_cond_t=zero_cond_t, num_layers=num_layers)
+    load_dit(model, dit_path, device=device, dtype=dtype, drop_keys=("__index_timestep_zero__"))
```
Comment (bitgamma) [inline thenoise/dit/qwen_image/models.py]: drop_keys=("__index_timestep_zero__") is a str, not a tuple (missing comma).

File: thenoise/dit/qwen_image/sampling.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    # terminal stretch would divide by zero; leave the grid as-is instead.
+    one_minus_z = 1 - sigmas
+    if one_minus_z[-1] > 0:
+        scale_factor = one_minus_z[-1] / (1 - 0.02)
+        sigmas = 1 - one_minus_z / scale_factor
+    return sigmas
+
+
+def compute_mu(image_seq_len: int) -> float:
+    from .utils import calculate_shift
```
Comment (bitgamma) [inline thenoise/dit/qwen_image/sampling.py]: let's move the import at top level. Also if compute_mu is only used in get_schedule, then just use calculate_shift directly there and remove this helper

File: thenoise/dit/qwen_image/utils.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+    if key.startswith("visual."):
+        return key.replace("visual.", "model.visual.", 1)
+    return key
+
+
+def load_qwen2_5_vl(
+    ckpt_path: str,
+    dtype: Optional[torch.dtype],
+    device: Union[str, torch.device],
+    disable_mmap: bool = True,
```
Comment (bitgamma) [inline thenoise/dit/qwen_image/utils.py]: this parameter is unused, remove

**Review by bitgamma — CHANGES_REQUESTED** [review]

**Review by bitgamma — APPROVED** [review]
lgtm
